src/client/gui/lib/vm_details/terminal_tabs.dart

```diff
@@ -1,10 +1,59 @@
 import 'package:basics/basics.dart';
+import 'package:built_collection/built_collection.dart';
 import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 
 import 'terminal.dart';
 
+typedef ShellIds = ({
+  BuiltList<int> ids,
+// this is the index of the currently selected shell id
+  int currentIndex,
+});
+
+class ShellIdsNotifier extends AutoDisposeFamilyNotifier<ShellIds, String> {
+  @override
+  ShellIds build(String arg) => (ids: [1].build(), currentIndex: 0);
+
+  void add() {
+    final ids = state.ids;
+    final newShellId = 1.to(1000).whereNot(ids.contains).first;
+    state = (
+      ids: ids.rebuild((ids) => ids.add(newShellId)),
+      currentIndex: ids.length,
+    );
+  }
+
+  void remove(int index) {
```

Comment:
After suspending, if there is only one terminal session, there are difficulties closing it; attempting to close the tab will initially do nothing and then it will close about 30 seconds later.

Reply:
Hmm, this is indeed a problem, and I see it happens in the CLI as well. Part of the problem is that when you suspend an instance, it doesn't close the SSH connection. On the contrary, even that gets suspended and restored once the instance is started again. So from the client side, it just looks like it takes some time for the instance to respond, but the connection is still active. Still, in the GUI we could manually close the connection from the client side if we see the instance is no longer running.

Reply:
Hmmm, or maybe not? Maybe it's a valid use-case for a user to be logged into a shell, suspend it for some reason, and resume it later and still have their SSH session intact?

Reply:
We could imagine doing that but closing the connection if/when the user closes the tab. However, I think there would need to be visual indication that the terminal is frozen, which would need design. We need to bring it up with them but, until we have that, I think it is safer to avoid the use-case entirely and just disconnect.